adam/perception/high_level_semantics_situation_to_developmental_primitive_perception.py

```diff
@@ -730,25 +773,27 @@ def _perceive_ground_relations(self):
                             ):
                                 object_is_on_something = True
                     if not object_is_on_something:
-                        self._relation_perceptions.append(
+                        ground_relations.append(
                             on(
                                 self._objects_to_perceptions[situation_object],
                                 perceived_ground,
-                            )
+                            )[0]
                         )
                 else:
-                    self._relation_perceptions.append(
+                    ground_relations.append(
                         on(
                             self._objects_to_perceptions[situation_object],
                             perceived_ground,
-                        )
+                        )[0]
                     )
 
+        return ground_relations
```

Comment:
Can you return the `ground_relations` in an `immutableset` I don't think passing out a list here is a big deal because it's immediately going to get chained into an `immutableset` before it would be iterated over but we prefer to return immutable structures for determinism purposes.